flink-libraries/flink-sql-client/src/main/java/org/apache/flink/table/client/config/Environment.java

```diff
@@ -29,38 +30,47 @@
 
 /**
  * Environment configuration that represents the content of an environment file. Environment files
- * define sources, execution, and deployment behavior. An environment might be defined by default or
+ * define tables, execution, and deployment behavior. An environment might be defined by default or
  * as part of a session. Environments can be merged or enriched with properties (e.g. from CLI command).
  *
  * <p>In future versions, we might restrict the merging or enrichment of deployment properties to not
  * allow overwriting of a deployment by a session.
  */
 public class Environment {
 
-	private Map<String, Source> sources;
+	private Map<String, TableDescriptor> tables;
```

Comment:
Why not maintaining two separate maps for sources and sinks? Then we don't need instance of checks. If a table is both we can simply add it to both maps.

Reply:
That is another option. But the purpose is if a table is both source and sink, we dont need to duplicate the config in both the sources and sinks section, as it might be error prone that you modify the table config in sources, but forget to modify the config of the same table in sinks section, thus causing inconsistency. What do you think?

Reply:
Also true. I will merge this now to have it in the next release. We can still refactor if necessary as it is internal API.